components/style/properties/shorthand/inherited_text.mako.rs

```diff
@@ -21,3 +21,63 @@
         }
     }
 </%helpers:shorthand>
+
+// https://drafts.csswg.org/css-text-decor-3/#text-emphasis-property
+<%helpers:shorthand name="text-emphasis" products="gecko" sub_properties="text-emphasis-color
+    text-emphasis-style">
+    use properties::longhands::{text_emphasis_color, text_emphasis_style};
+
+    pub fn parse_value(context: &ParserContext, input: &mut Parser) -> Result<Longhands, ()> {
+        let mut color = None;
+        let mut style = None;
+
+        try!(input.try(|input| {
+            loop {
+                if color.is_none() {
+                    if let Ok(value) = input.try(|input| text_emphasis_color::parse(context, input)) {
+                        color = Some(value);
+                        continue
+                    }
+                }
+                if style.is_none() {
+                    if let Ok(value) = input.try(|input| text_emphasis_style::parse(context, input)) {
+                        style = Some(value);
+                        continue
+                    }
+                }
+                break
+            }
+            if color.is_some() || style.is_some() {
+                if style.is_none() {
+                    style = Some(text_emphasis_style::get_initial_specified_value());
+                }
+                Ok(())
+            } else {
+                Err(())
+            }
+        }));
+
+        Ok(Longhands {
+            text_emphasis_color: color,
+            text_emphasis_style: style,
+        })
+    }
+
+    impl<'a> LonghandsToSerialize<'a>  {
+        fn to_css_declared<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            if let DeclaredValue::Value(ref value) = *self.text_emphasis_style {
+                try!(value.to_css(dest));
+            } else {
+                try!(write!(dest, "none"));
+            }
+
+            try!(write!(dest, " "));
+
+            if let DeclaredValue::Value(ref color) = *self.text_emphasis_color {
+                color.to_css(dest)
+            } else {
+                write!(dest, "currentColor")
```

Comment:
Yeah, I noticed in my CSSOM work that we output currentColor in a bunch of places. This needs to be fixed all over servo style (can be done separately I guess)